compiler/rustc_session/src/options.rs

```diff
@@ -2114,6 +2123,9 @@ written to standard error output)"),
     #[rustc_lint_opt_deny_field_access("use `Session::tls_model` instead of this field")]
     tls_model: Option<TlsModel> = (None, parse_tls_model, [TRACKED],
         "choose the TLS model to use (`rustc --print tls-models` for details)"),
+    #[rustc_lint_opt_deny_field_access("use `Session::tls_model` instead of this field")]
+    tls_dialect: Option<TlsDialect> = (None, parse_tls_dialect, [TRACKED],
```

Comment:
Maybe name this `-Zuse-tlsdesc` and have it take a bool? I don't suppose we get another dialect before the transition from traditional to tlsdesc TLS support is finished, right?

Reply:
I'm OK w/ that, but I think its usually better to support an existing spelling when there are existing conventions in other compilers. We'd have to check w/ folks doing the GCC backend work, but its possible the variants would be easier to integrate for them.

We've talked about adding support for a 3 GOT slot variant of TLSDESC. That would require support from the linker and libc.  There are advantages to doing it that way, like avoiding a level of indirection to read the descriptor for the dynamic cases, but I'm not sure how likely that is to ever happen. I'd like to see it, and was hoping that would be the ABI for RISC-V, but it would be a win for any architecture.

But I can change this to a boolean if you'd like.